march_hardware/test/TestJoint.cpp

```diff
@@ -149,3 +149,166 @@ TEST_F(JointTest, GetTemperatureWithoutTemperatureGes)
   march::Joint joint("get_temperature", 0, false, nullptr, nullptr);
   ASSERT_FLOAT_EQ(joint.getTemperature(), -1.0);
 }
+
+TEST_F(JointTest, TestPrepareActuation)
+{
+  EXPECT_CALL(*this->imc, getAngleRadIncremental()).WillOnce(Return(5));
+  EXPECT_CALL(*this->imc, getAngleRadAbsolute()).WillOnce(Return(3));
+  EXPECT_CALL(*this->imc, goToOperationEnabled()).Times(1);
+  march::Joint joint("actuate_true", 0, true, std::move(this->imc));
+  joint.prepareActuation();
+  ASSERT_EQ(joint.getIncremental_position(), 5);
+  ASSERT_EQ(joint.getAbsolute_position(), 3);
+}
+
+TEST_F(JointTest, TestReceivedDataUpdateFirstTimeTrue)
+{
+  EXPECT_CALL(*this->imc, getMotorVoltage()).WillOnce(Return(48));
+  march::Joint joint("actuate_true", 0, true, std::move(this->imc));
+  ASSERT_TRUE(joint.receivedDataUpdate());
+}
+
+TEST_F(JointTest, TestReceivedDataUpdateTrue)
+{
+  EXPECT_CALL(*this->imc, getMotorVoltage()).WillOnce(Return(48)).WillOnce(Return(48.001));
+  march::Joint joint("actuate_true", 0, true, std::move(this->imc));
+  joint.receivedDataUpdate();
+  ASSERT_TRUE(joint.receivedDataUpdate());
+}
+
+TEST_F(JointTest, TestReceivedDataUpdateFalse)
+{
+  EXPECT_CALL(*this->imc, getMotorVoltage()).WillRepeatedly(Return(48));
+  march::Joint joint("actuate_true", 0, true, std::move(this->imc));
+  joint.receivedDataUpdate();
+  ASSERT_FALSE(joint.receivedDataUpdate());
+}
+
+TEST_F(JointTest, TestReadEncodersOnce)
+{
+  ros::Duration elapsed_time(0.2);
+  double velocity = 0.5;
+  double absolute_noise = -2 * this->imc->getAbsoluteRadPerBit();
+
+  double initial_incremental_position = 5;
+  double initial_absolute_position = 3;
+  double new_incremental_position = initial_incremental_position + velocity * elapsed_time.toSec();
+  double new_absolute_position = initial_absolute_position + velocity * elapsed_time.toSec() + absolute_noise;
+
+  EXPECT_CALL(*this->imc, getMotorVoltage()).WillOnce(Return(48));
+  EXPECT_CALL(*this->imc, getAngleRadIncremental())
+      .WillOnce(Return(initial_incremental_position))
+      .WillOnce(Return(new_incremental_position));
+  EXPECT_CALL(*this->imc, getAngleRadAbsolute())
+      .WillOnce(Return(initial_absolute_position))
+      .WillOnce(Return(new_absolute_position));
+
+  march::Joint joint("actuate_true", 0, true, std::move(this->imc));
+  joint.prepareActuation();
+
+  joint.readEncoders(elapsed_time);
+
+  ASSERT_NEAR(joint.getPosition(), initial_absolute_position + velocity * elapsed_time.toSec(), std::pow(10, -9));
+  ASSERT_NEAR(joint.getVelocity(), velocity, std::pow(10, -9));
```

Comment:
There also exists a [`ASSERT_DOUBLE_EQ`](https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#floating-point-comparison) which will check `double`s for equality within 4 Units in the Last Place (ULPs). Which will work for better for all value ranges and you don't have to give an error bound.

Reply:
Thanks that makes things better :)